src/components/SummaryCard/__tests__/SummaryCard.spec.js

```diff
@@ -0,0 +1,178 @@
+/**
+ * @file Summary card tests.
+ * @copyright IBM Security 2020
+ */
+
+import Folder20 from '@carbon/icons-react/lib/folder/20';
+import { render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+
+import {
+  SummaryCard,
+  SummaryCardHeader,
+  SummaryCardBody,
+  SummaryCardAction,
+  SummaryCardFooter,
+  Tooltip,
+} from '../../..';
+
+import { carbonPrefix } from '../../../globals/namespace';
+import { namespace as headerNamespace } from '../SummaryCardHeader/SummaryCardHeader';
+
+describe('SummaryCard', () => {
+  test('should have no Axe or DAP violations`', async () => {
```

Comment:
I'm torn between whether we should be testing the a11y of this component and sub-components as a whole or in individual unit tests - personally, it feels right to do all of these individually, but I'd like to get your thoughts!

Reply:
The `SummaryCard` a11y test would really mean nothing without the subcomponents included.
And to test the subcomponents individually, after having already included them together as intended inside the `SummaryCard` a11y test, seems redundant to me. 

What are you seeking to gain by rendering and testing each subcomponent individually, rather than doing it this way?

Reply:
The distinction here is that as we're including these with our unit tests, my assumption is that we would be testing the accessibility of each individual component or unit.

If we want to test them as a whole, I believe that's more in line with integration testing, and we should define a strategy of how we want to proceed with that.

Reply:
Because the `SummaryCard` subcomponents depend on the structure of the parent `SummaryCard` and would not be used elsewhere, I agree that it does not make sense to test each subcomponent individually. As @jendowns said, the subcomponents will lose their semantic meaning without living inside of the `SummaryCard`.

To handle these types of situations in DE, we usually have a `Component.test.js` file for unit tests and a `Component.integration.test.js` file for integration tests that test how interacting with one component might alter the behavior of another (usually a child affecting its parent)

Reply:
I agree it's pointless to unit test individually if we want to verify overall semantics - if we have one component that's a `ul` and a subcomponent that's a `div`, we won't identify that this is an issue if these aren't integrated.

I'm trying to tease out the best approach here, as there are a number of options I can think of off the top of my head - These include:

- having dedicated integration tests
- the catch-all approach in `SummaryCard.spec.js`
- an approach where the `a11y` tests for sub-components live as a hybrid integration / unit test in their own testing suite (for example `SummaryCardAction.spec.js`) and require the components needed to verify the semantics (potentially `SummaryCard` and `SummaryCardAction`)

The a11y tests are unique in that we're validating the rendered output, so it's harder for us to point to any methodology and say 'this is the right way to do it' - I'm definitely not suggesting we go with one or the other, more so that we should consider and discuss the possibilities before making a call on it.

Reply:
Considering the above, what are you seeking to change here that is currently missing / insufficient? 

I think the conversation is getting very meta now, and we should step back and evaluate the effectiveness of the code.

If the test was broken down and split, that could make it less effective (as you described yourself with the UL and DIV example, and as Diego and I also described above).

If the test is left as-is... then it remains effective, right?

In that case, can we make a final call based on the effectiveness of the test? It appears to be effective, and there are no more effective alternatives.

Reply:
Considering the above, what are you seeking to change here that is currently missing / insufficient? 

I think the conversation is getting very meta now, and we should step back and evaluate the effectiveness of the code.

If the test was broken down and split, that could make it less effective (as you described yourself with the UL and DIV example, and as Diego and I also described above).

If the test is left as-is... then it remains effective, right?

In that case, can we make a final call based on the effectiveness of the test? It appears to be effective, and there are no more effective alternatives.